### PR TITLE
Update with NHS Board data tab and weekly trends chart

### PR DESCRIPTION
**Pull request by calump96**

### CODE REVIEW

File: shiny_app/data_tab.R [hunk 1]
```diff
@@ -6,18 +6,21 @@ data_table <- reactive({  # Change dataset depending on what user selected
                        "tidyLFT" = tidyLFT %>% rename (`Profession` = test_cohort_name, 
                                                        `Work location type` = LocationType,
                                                        `Work location name` = LocationName, 
+                                                       `NHS Board` = Health_Board_Name, 
                                                        `Test result` = test_result, 
                                                        `Date` = new_date),
-                       "TestNumbers" = TestNumbers %>%  rename (`Week ending` = roll_week_ending, 
+                       "TestNumbers" = TestNumbers %>%  rename (`Week ending` = week_ending,
```
Comment (calump96) [inline shiny_app/data_tab.R:12]: Making the data tab weekly rather than using the rolling four week data

File: shiny_app/functions_server.R [hunk 2]
```diff
@@ -57,8 +58,8 @@ plot_location_chart <- function(dataset, data_name, settingdata, yaxis_title, ar
 
   trend_data <- tidyLFT %>% 
     filter(test_cohort_name %in% input$Profession_select) %>% 
-    filter(LocationName %in% input$Location_select)  %>%
```
Comment (calump96) [inline shiny_app/functions_server.R:60]: Use Health Board rather than Location

File: shiny_app/functions_server.R [hunk 3]
```diff
@@ -68,14 +69,14 @@ plot_location_chart <- function(dataset, data_name, settingdata, yaxis_title, ar
...
   # ext for tooltip
   tooltip_trend <- glue("{trend_data$test_cohort_name}<br>",
-                        "{trend_data$LocationType}: {trend_data$LocationName}<br>",
+                        "{trend_data$Health_Board_Name}<br>",
                         "Number of {trend_data$test_result} tests: {trend_data$Count}")
 
   # Creating contact tracing time
   trend_data %>%
-    plot_ly(x = ~LocationName, y = ~Count) %>%
+    plot_ly(x = ~Health_Board_Name, y = ~Count) %>%
```
Comment (calump96) [inline shiny_app/functions_server.R:77]: Use Health Board rather than Location for charts

File: shiny_app/functions_server.R [hunk 4]
```diff
@@ -95,24 +96,64 @@ plot_testnumbers_chart <- function(dataset, data_name, settingdata, yaxis_title,
   
   trend_data <- TestNumbersChart %>%
```
Comment (calump96) [inline shiny_app/functions_server.R:97]: Add chart for week ending

File: shiny_app/global.R [hunk 5]
```diff
@@ -30,10 +30,12 @@ library(shinycssloaders) #for loading icons, see line below
 ############################################### Data ###############################################
 
 tidyLFT <-readRDS("data/tidyLFT.rds")
-LabCases <-readRDS("data/LabCases.rds")
 CHICapture <-readRDS("data/CHICapture.rds")
 TestNumbers <- readRDS("data/test_numbers.rds")
 TestNumbersChart <- readRDS("data/test_numbers_chart.rds")
+TestNumbersChartRoll <- readRDS("data/roll_test_numbers_chart.rds")
```
Comment (calump96) [inline shiny_app/global.R:36]: New files required for running dashboard

File: shiny_app/global.R [hunk 6]
```diff
@@ -67,16 +69,28 @@ plot_cut_missing <- function(title_plot, plot_output, extra_content = NULL) {
 #Create a function which creates all unique  
 Profession <- c(sort(unique(tidyLFT$test_cohort_name)))
 Work_Location <- c(unique(tidyLFT$LocationName))
+LFT_Profession <- c(sort(unique(LFT_hb$test_cohort_name)))
+Time_period <- c(sort(unique(LFT_hb$time_period)))
+LFT_NHS_Board <- c(sort(unique(LFT_hb$Health_Board_Name)))
```
Comment (calump96) [inline shiny_app/global.R:74]: New drop down choices

File: shiny_app/global.R [hunk 7]
```diff
@@ -67,16 +69,28 @@ plot_cut_missing <- function(title_plot, plot_output, extra_content = NULL) {
...
-pal_overall <- c('#000000', '#0078D4','#9B4393', '#bdbdbd', '#bdbdbd', '#bdbdbd', '#7fcdbb')
+# pal_overall <- c('#9B4393', '#0078D4','#000000', '#bdbdbd', '#bdbdbd', '#bdbdbd', '#7fcdbb')
+
+pal_tests <- c("NEGATIVE" = "#0078D4", "INCONCLUSIVE" = "#9B4393", 
+               "POSITIVE" = "#000000")
+
+pal_n_tests <- c("1" = "#082359", "2" = "#0078D4", "3" = "#9B4393", 
+                 "4" = "#dcb4e9", "5" = "#e9b4e0", "6" = "#00cdd4",
+                 "7" = "#7fcdbb", "8" = "#7fcd99", "9" = "#94cd7f", 
+                 "10+" = "#000000")
```
Comment (calump96) [inline shiny_app/global.R:93]: New colour palettes for charts

File: shiny_app/hb_tab.R [hunk 8]
```diff
@@ -0,0 +1,133 @@
+############################################### Modals ###############################################
```
Comment (calump96) [inline shiny_app/hb_tab.R:1]: Code for new HB tab

File: shiny_app/server.R [hunk 9]
```diff
@@ -16,6 +16,10 @@ function(input, output, session) {
   # Sourcing file with functions code
   source(file.path("functions_server.R"),  local = TRUE)$value
   
+  ###############################################.
+  # NHS Boards data tab  
+  source(file.path("hb_tab.R"),  local = TRUE)$value
```
Comment (calump96) [inline shiny_app/server.R:21]: Use new HB tab

File: shiny_app/summary_tab.R [hunk 10]
```diff
@@ -18,11 +18,11 @@ observeEvent(input$btn_dataset_modal,
 
 observeEvent(input$Profession_select, {
 
-  data <- tidyLFT %>%
+  data <- LFT_hb %>%
```
Comment (calump96) [inline shiny_app/summary_tab.R:21]: Use HB data

**Review by jadecarruthers — APPROVED** [review]

**Review by jadecarruthers — APPROVED** [review]
Happy

**Comment by jadecarruthers** [comment]
Happy

**Pull request merged**